includes/data-stores/class-wcs-orders-table-subscription-data-store.php

```diff
@@ -422,6 +422,16 @@ public function untrash_order( WC_Order $subscription ): bool {
 				wp_untrash_post( $id );
 			}
 
+			/**
+			 * Fires after a subscription is restored from the trash.
+			 *
+			 * @since 5.2.0
+			 *
+			 * @param int    $subscription_id Subscription ID.
+			 * @param string $previous_status The status of the subscription before it was trashed.
+			 */
+			do_action( 'woocommerce_subscription_untrashed', $subscription->get_id(), $previous_status );
```

Comment:
Yes. Since there is a possibility that untrashing might fail, I try to hook it after but you're right that it won't cover order, only subscription.

Do you think that's the best we can do, since there is no hook that fires after untrashing?

Reply:
> Do you think that's the best we can do, since there is no hook that fires after untrashing?

yeah I guess that's the best we can do. We will just have to assume that in most cases, restoring the order/subscription will be successful.